module/VuFind/src/VuFind/View/Helper/Root/Flashmessages.php

```diff
@@ -82,7 +82,13 @@ public function __invoke()
                 $this->fm->getMessages($ns), $this->fm->getCurrentMessages($ns)
             );
             foreach (array_unique($messages, SORT_REGULAR) as $msg) {
-                $html .= '<div class="' . $this->getClassForNamespace($ns) . '">';
+                $html .= '<div class="' . $this->getClassForNamespace($ns) . '"';
+                if (isset($msg['dataset'])) {
+                    foreach ($msg['dataset'] as $attr => $value) {
+                        $html .= ' data-' . $attr . '="' . $value . '"';
```

Comment:
Should we be escaping $attr and/or $value here? Seems like htmlspecialchars() might be in order to be on the safe side.